WooCommerce/WooCommerceTests/Tools/Observables/BehaviorSubjectTests.swift

```diff
@@ -0,0 +1,329 @@
+        }
+
+        // When
+        subject.send("dicta")
+        subject.send("amet")
+        subject.send("dolor")
+
+        // Then
+        XCTAssertEqual(emittedValues, ["exercitationem", "dicta", "amet", "dolor"])
+    }
+
+    func test_it_will_emit_values_to_all_observers() {
+        // Given
+        let subject = BehaviorSubject("provident")
+
+        var emittedValues = [String]()
+
+        // First observer
+        _ = subject.subscribe { value in
+            emittedValues.append(value)
+        }
+        // Second observer
+        _ = subject.subscribe { value in
+            emittedValues.append(value)
+        }
+
+        // When
+        subject.send("dicta")
+        subject.send("amet")
+
+        // Then
+        // We'll receive two values for each observer
+        XCTAssertEqual(emittedValues, ["provident", "provident", "dicta", "dicta", "amet", "amet"])
+    }
+
+    func test_it_will_not_emit_values_to_cancelled_observers() {
+        // Given
+        // "dicta" will be emitted because it is the initial value
+        let subject = BehaviorSubject("dicta")
+
+        var emittedValues = [String]()
+        let observationToken = subject.subscribe { value in
+            emittedValues.append(value)
+        }
+
+        // When
+        // Cancel the observer
+        observationToken.cancel()
+
+        // This will not be emitted anymore
+        subject.send("amet")
+
+        // Then
+        XCTAssertEqual(emittedValues, ["dicta"])
+    }
+
+    // MARK: - Proof of Compatibility with Combine's CurrentValueSubject
+
+    func test_Combine_CurrentValueSubject_keeps_the_initial_value() throws {
+        guard #available(iOS 13.0, *) else {
+            try XCTSkipIf(true, "This test is for iOS 13.0+ only")
+            return
+        }
+
+        let subject = CurrentValueSubject<String, Error>("dolorem")
+
+        XCTAssertEqual(subject.value, "dolorem")
+    }
+
+    func test_Combine_CurrentValueSubject_keeps_the_last_emitted_value() throws {
+        guard #available(iOS 13.0, *) else {
+            try XCTSkipIf(true, "This test is for iOS 13.0+ only")
+            return
+        }
+
+        // Given
+        let subject = CurrentValueSubject<String, Error>("dolorem")
+
+        // When
+        subject.send("autem")
+
+        // Then
+        XCTAssertEqual(subject.value, "autem")
+    }
+
+    func test_Combine_CurrentValueSubject_will_emit_the_initial_value_to_an_observer() throws {
+        guard #available(iOS 13.0, *) else {
+            try XCTSkipIf(true, "This test is for iOS 13.0+ only")
+            return
+        }
+
+        // Given
+        let subject = CurrentValueSubject<String, Error>("dolorem")
+        var disposeBag = Set<AnyCancellable>()
+
+        // When
+        var emittedValues = [String]()
+        subject.sink(receiveCompletion: { _ in
+            // noop
+        }) { value in
+            emittedValues.append(value)
+        }.store(in: &disposeBag)
```

Comment:
dispose bag could be an eyesore 🙄

Reply:
I learned that I can use `cancellables` for the name. But if you meant adding all these variables, then I have no good answer. 😄 We can probably find something to make them seamless.